neurodiffeq/solvers.py

```diff
@@ -1357,7 +1357,7 @@ def __init__(self, ode_system, conditions, t_min, t_max,
 
         def _diff_eqs_wrapper(*variables):
             funcs_and_coords = variables[:N_FUNCTIONS + N_COORDS]
-            eq_params = tuple(variables[idx] for idx in eq_param_index)
+            eq_params = tuple(variables[idx] for idx in self.eq_param_index)
```

Comment:
Is there a particular reason we want to use `self.eq_param_index` instead of `eq_param_index` here? Both ways should have the same effect, but I don't like `self.diff_eqs` to be a closure that depends on self. This would make certain operations complicated, such as recreating a solver 
```python
solver1 = BundleSolver1D(..., eq_param_index1)
solver2 = BundleSolver1D(..., eq_param_index2)
solver2.diff_eqs = solver1.diff_eqs
```
If `eq_param_index1` and `eq_param_index2` are different, the above won't work, as `solver1.diff_eqs` will always depend on solver1.

Reply:
hmm.. my motivation to use `self.eq_param_index` was to use this internal variable when loading a saved solver as otherwise we have to recompute `eq_param_index` again and pass it in the constructor.